js/search/engine.js

```diff
@@ -249,6 +232,18 @@ let get_default = function () {
         the_engine = new Engine({
             language: language,
         });
+
+        Object.defineProperty(the_engine, 'default_app_id', {
+            set: function (id) {
```

Comment:
I don't think this needs to be dynamic. I think this can just be a setter on the engine class. And if we don't want it to be called during the constructor then just don't include ` GObject.ParamFlags.CONSTRUCT` in the property definition